core/authn.go

```diff
@@ -34,26 +36,56 @@ type tokenResult struct {
 
 func (a *apiAuthn) handler(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
-		err := a.auth(req)
-		if err != nil {
-			errorFormatter.Write(req.Context(), rw, err)
+		token, err0 := a.tokenAuth(req)
+		ctx := req.Context()
+		if err0 == nil {
+			// if this request was successfully authenticated, pass the token along
+			if token != "" {
```

Comment:
Suggest combining this `if` with the one above (via `&&`) to avoid unnecessary `if` nesting.